src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -1310,31 +1310,15 @@ public void refreshPosition(@NotNull final Pos newPosition) {
     }
 
     /**
-     * @return The height offset for passengers of this vehicle
-     */
-    private double getPassengerHeightOffset() {
-        // TODO: Move this logic elsewhere
-        if (entityType == EntityType.BOAT) {
-            return -0.1;
-        } else if (entityType == EntityType.MINECART) {
-            return 0.0;
-        } else {
-            return entityType.height() * 0.75;
-        }
-    }
-
-    /**
-     * Sets the X,Z coordinate of the passenger to the X,Z coordinate of this vehicle
-     * and sets the Y coordinate of the passenger to the Y coordinate of this vehicle + {@link #getPassengerHeightOffset()}
+     * Sets the coordinates of the passenger to the coordinates of this vehicle + {@link EntityUtils#getPassengerHeightOffset(Entity, Entity)}
      *
-     * @param newPosition The X,Y,Z position of this vehicle
-     * @param passenger   The passenger to be moved
+     * @param newPosition the new position of this vehicle
+     * @param passenger   the passenger to be moved
      */
     private void updatePassengerPosition(Point newPosition, Entity passenger) {
         final Pos oldPassengerPos = passenger.position;
-        final Pos newPassengerPos = oldPassengerPos.withCoord(newPosition.x(),
-                newPosition.y() + getPassengerHeightOffset(),
-                newPosition.z());
+        final Pos newPassengerPos = oldPassengerPos.withCoord(newPosition.x(), newPosition.y(), newPosition.z())
```

Comment:
Shouldn't the mutation be done in a single `withCoord` call like before? This version isn't more readable and does one more allocation

Reply:
> Shouldn't the mutation be done in a single `withCoord` call like before? This version isn't more readable and does one more allocation

Yeah, that was a mistake from reverting the big change to passenger offsets I had initially made